cmd/update-main.go

```diff
@@ -98,26 +97,39 @@ func GetCurrentReleaseTime() (releaseTime time.Time, err error) {
 	return getCurrentReleaseTime(Version, os.Args[0])
 }
 
-func isDocker(cgroupFile string) (bool, error) {
-	cgroup, err := ioutil.ReadFile(cgroupFile)
-	if os.IsNotExist(err) {
-		err = nil
+// Check if we are indeed inside docker.
+// https://github.com/moby/moby/blob/master/daemon/initlayer/setup_unix.go#L25
+//
+//     "/.dockerenv":      "file",
+//
+func isDocker(dockerEnvFile string) (ok bool, err error) {
+	_, err = os.Stat(dockerEnvFile)
+	if err != nil {
+		if os.IsNotExist(err) {
+			err = nil
+		}
+		return false, err
 	}
-
-	return bytes.Contains(cgroup, []byte("docker")), err
+	return true, nil
 }
 
 // IsDocker - returns if the environment minio is running
 // is docker or not.
 func IsDocker() bool {
-	found, err := isDocker("/proc/self/cgroup")
-	fatalIf(err, "Error in docker check.")
+	found, err := isDocker("/.dockerenv")
+	// We don't need to fail for this check, log
+	// an error and return false.
+	errorIf(err, "Error in docker check.")
 
 	return found
 }
 
-// IsKubernetes returns if the environment minio is
-// running is kubernetes or not.
+// IsDCOS returns true if minio is running in DCOS.
+func IsDCOS() bool {
+	return os.Getenv("MARATHON_APP_LABEL_DCOS_PACKAGE_NAME") == "minio"
```

Comment:
Is this a stable interface? Please add the relevant document describing the above environment variable usage in comments.